mtraining/src/main/java/org/motechproject/mtraining/domain/CourseUnitMetadata.java

```diff
@@ -82,4 +86,37 @@ public String getDescription() {
     public void setDescription(String description) {
         this.description = description;
     }
+
+    /**
+     * Builds the dto representation of the unit used by the tree view.
+     *
+     * @return the dto representation of the unit
+     */
+    @Ignore
+    public CourseUnitDto toUnitDto() {
+        return new CourseUnitDto(getId(), getName(), getState().toString(), null, null);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        final CourseUnitMetadata other = (CourseUnitMetadata) obj;
+
+        return Objects.equals(this.getId(), other.getId())
+                && Objects.equals(this.getName(), other.getName())
+                && Objects.equals(this.getContent(), other.getContent())
+                && Objects.equals(this.getDescription(), other.getDescription());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId());
```

Comment:
not consistent with equals, also the use of ID field should be avoided in the hashCode method